health/health.c

```diff
@@ -215,7 +215,7 @@ static inline void health_alarm_execute(RRDHOST *host, ALARM_ENTRY *ae) {
         else {
             // we have not executed this alarm notification in the past
             // so, don't send CLEAR notifications
-            if(unlikely(ae->new_status == RRDCALC_STATUS_CLEAR)) {
+            if(unlikely(ae->new_status == RRDCALC_STATUS_CLEAR) && (ae->old_status < RRDCALC_STATUS_RAISED)) {
```

Comment:
This doesn't quite do the trick, see below. We're changing a code used by non-repeating alarm notifications, because we don't maintain the notifications history in the structure `ae` (i.e. `ae->next` is always null). Perhaps this whole part of the code (including the loop above where `t` iterates through `ae->next`) needs to be different for repeating alarms. 

Not sure why we got repeating CLEAR alarm notifications, but it's certainly not what we want. 
 
![Screenshot from 2019-08-20 12-17-13](https://user-images.githubusercontent.com/43294513/63339449-31f7e480-c345-11e9-9254-959e59e4b0e5.png)

Reply:
Firstly thank you, I saw this result and I also did not like the result, but I was without an idea about how to get what we want, with your comments I had an idea that I will try.

Reply:
I changed the rule to display the clear alarm when we have the alarms repeating enabled, I decided to create flags to help us to display the message when it is necessary, this does not affect the normal routine, because the newest flag  is enabled only when the repeat feature is enabled.

Reply:
Please test with the following alarms, after configuring to receive slack notifications. example1 one is repeating, example2 is not. When you see it working well for several minutes (with alarms getting raised and being cleared for both), I'll test as well.

```
   alarm: example1
      on: example.random
  lookup: average -10s of random1
    warn: $this > (($status >= $WARNING)  ? (50) : (50))
    crit: $this > (($status == $CRITICAL) ? (60) : (60))
    info: random
      to: admin
every: 1s
repeat: warning 5s critical 5s

   alarm: example2
      on: example.random
  lookup: average -10s of random2
    warn: $this > (($status >= $WARNING)  ? (50) : (50))
    crit: $this > (($status == $CRITICAL) ? (60) : (60))
    info: random
      to: admin
every: 1s
```

Reply:
Hi @cakrit ,

I tested today during various hours with this alarm that you gave me, I generated more than 1500 alarms and I could not see clear alarm repeating, please, can you confirm that this PR is ok? 
An additional information, I tested the alarm  not setting the global option for repeating and with the global option for repeating alarm.

Reply:
Tested and it's good.